formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -125,25 +175,39 @@ internal class FormulaManagerImpl<Input, State, Output>(
         }
 
         val frame = Frame(snapshot, result, transitionID)
-        actionManager.onNewFrame(frame.evaluation.actions)
         this.frame = frame
 
+        actionManager.onNewFrame(frame.evaluation.actions)
```

Comment:
What are the ramifications and/or intentions of moving this line? There seems to be a number of affected codepaths in evaluation if `actionManager.onNewFrame` caused re-evaluations.

Reply:
No ramifications, it doesn't perform any side-effecting logic. It only invalidates internal ActionManager state which will be picked up in `FormulaManagerImpl.executeUpdates`